x-pack/legacy/plugins/ml/public/application/datavisualizer/file_based/components/filebeat_config_flyout/filebeat_config.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { FindFileStructureResponse } from '../../../../../../common/types/file_datavisualizer';
+
+export function createFilebeatConfig(
+  index: string,
+  results: FindFileStructureResponse,
+  ingestPipelineId: string,
+  username: string | null
+) {
+  return [
+    'filebeat.inputs:',
+    `- type: ${index}`,
+    '  paths:',
+    "  - '<add path to your files here>'",
+    ...getEncoding(results),
+    ...getExcludeLines(results),
+    ...getMultiline(results),
+    '',
+    ...getProcessors(results),
+    'output.elasticsearch:',
+    '  hosts:["<es_url>"]',
+    ...getUserDetails(username),
+    `  index: "${index}"`,
+    `  pipeline: "${ingestPipelineId}"`,
+  ].join('\n');
+}
+
+function getEncoding(results: any) {
+  return results.charset !== 'UTF-8' ? [`  encoding: ${results.charset}`] : [];
+}
+
+function getExcludeLines(results: any) {
+  return results.exclude_lines_pattern !== undefined
+    ? [`  exclude_lines: ['${results.exclude_lines_pattern}']`]
```

Comment:
If `results.exclude_lines_pattern` contains any single quote characters then they need doubling up.  Something along the lines of `results.exclude_lines_pattern.replace(/'/, "''")`.

Reply:
Fixed in 96de6a22e4484f4ddb9f0b01db4e59713a4408db